packages/blade/src/components/Popover/_decisions/decisions.md

```diff
@@ -0,0 +1,275 @@
+# Popover API Decisions <!-- omit in toc -->
+
+The popover component is used to provide context as well as enable users to take certain actions on it. They are designed to display relevant information without requiring the user to navigate to a different page or dialog. Popovers are triggered on click or onload.
+
+<img src="./assets/popover-thumbnail.png" width="380" alt="Popover Thumbnail" />
+
+- [Design](#design)
+- [Basic Features](#basic-features)
+- [`Popover` API](#popover-api)
+  - [Examples:](#examples)
+    - [Usage](#usage)
+    - [Controlled](#controlled)
+    - [Uncontrolled](#uncontrolled)
+    - [On initial load](#on-initial-load)
+    - [Working with non-interactive triggers](#working-with-non-interactive-triggers)
+  - [Custom Triggers](#custom-triggers)
+- [Future Scope](#future-scope)
+- [Tooltips vs Popovers vs Guided Popovers](#tooltips-vs-popovers-vs-guided-popovers)
+- [Library](#library)
+- [Motion](#motion)
+- [Accessibility](#accessibility)
+- [Open Question](#open-question)
+- [References](#references)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/LSG77hEeVYDk7j7WV7OMJE/Blade-DSL---Components-Guideline?type=design&node-id=2875-40870&mode=design&t=mTbpUCQOr7kHk6UT-0) to all variants of the Popover component
+
+## Basic Features
+
+- Controlled & Uncontrolled state
+- Can be triggered on click or on load
+- Can be placed on different placements (top, bottom, left, right)
+
+## `Popover` API
+
+| Prop            | Type                                                                                                                   | Default     | Description                                                                                                                           | Required |
+| --------------- | ---------------------------------------------------------------------------------------------------------------------- | ----------- | ------------------------------------------------------------------------------------------------------------------------------------- | -------- |
+| headerTitle     | `string`                                                                                                               | `undefined` | Title of the header                                                                                                                   | ✅        |
```

Comment:
We could do that, but we also have leading prop, and the header itself is a separate section inside Popover, prefixing with "header" helps with the intent that the "title" prop is for header's title, because in some design system they use the "title" prop as the content of the popover. (eg: [mui](https://mui.com/material-ui/react-tooltip/#basic-tooltip))

Reply:
In Blade's terminology I think `title` and `description` sound better than `headerTitle` and `content`. The title in the Modal is also called `title` and not `headerTitle` even if it appears in the ModalHeader primarily because it defines the content inside the modal.

`headerLeading` is also an unclear name on what the API is supposed to do. Depending on what the vocabulary is we can also call it `titleLeadingAsset` or `headerLeadingAsset`.

Reply:
agree with abinash on this

Reply:
Okay it seems we will need a larger discussion around this, lets discuss this tommorow.

Reply:
Also, noticed - we have followed the same naming conventions in Card and BottomSheet - `title` in `ComponentHeader`. Need to follow a similar convention, don't think we have ever had a prop called `headerTitle` yet.

Reply:
Discussed, going ahead without prefixing. 

headerTitle -> title
headerLeading -> titleLeading